### PR TITLE
Add AllWasmCodeID query

### PR DESCRIPTION
**Pull request by vuong177**
## Description
Currently, the  `codeID`  is a bit hard to track, we can only find `codeID` by querying `MsgPushNewWasmCode` transaction.
This PR is for adding a new `AllWasmCodeIDs` querier to query all codeID of contracts in the 08-wasm state.

### CODE REVIEW

File: modules/light-clients/08-wasm/keeper/msg_server.go [hunk 1]
```diff
@@ -18,9 +17,9 @@ var _ types.MsgServer = Keeper{}
 func (k Keeper) PushNewWasmCode(goCtx context.Context, msg *types.MsgPushNewWasmCode) (*types.MsgPushNewWasmCodeResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	if k.authority != msg.Signer {
-		return nil, sdkerrors.Wrapf(govtypes.ErrInvalidSigner, "invalid authority: expected %s, got %s", k.authority, msg.Signer)
-	}
+	// if k.authority != msg.Signer {
+	// 	return nil, sdkerrors.Wrapf(govtypes.ErrInvalidSigner, "invalid authority: expected %s, got %s", k.authority, msg.Signer)
+	// }
```
Comment (misko9) [inline modules/light-clients/08-wasm/keeper/msg_server.go:23]: Unfortunately, this is necessary and a few months ago, became a requirement for contract uploads to be restricted to governance proposals due to their sensitive access in ibc-go.

File: go.mod [hunk 2]
```diff
@@ -24,12 +24,18 @@ require (
...
 	github.com/stretchr/testify v1.8.2
-	google.golang.org/genproto v0.0.0-20230216225411-c8e22ba71e44
+	google.golang.org/genproto v0.0.0-20230306155012-7f2fa6fef1f4
 	google.golang.org/grpc v1.53.0
-	google.golang.org/protobuf v1.29.1
+	google.golang.org/protobuf v1.30.0
 	gopkg.in/yaml.v2 v2.4.0
 )
 
+require (
```
Comment (misko9) [inline go.mod:33]: nit, can this be in the same require block as the other indirects?

File: modules/light-clients/08-wasm/client/cli/query.go [hunk 3]
```diff
@@ -47,3 +47,43 @@ func getCmdCode() *cobra.Command {
...
+		),
+		Args: cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			queryClient := types.NewQueryClient(clientCtx)
+
+			pageReq, err := client.ReadPageRequest(cmd.Flags())
```
Comment (misko9) [inline modules/light-clients/08-wasm/client/cli/query.go:68]: What are the flags for pagination? I don't see them in the cli help.

**Review by misko9 — CHANGES_REQUESTED** [review]
Thanks for the PR! Let me know if you have any questions on the comments!

File: modules/light-clients/08-wasm/types/validation.go [hunk 4]
```diff
@@ -1,6 +1,6 @@
 package types
 
-var MaxWasmSize = 3 * 1024 * 1024
+var MaxWasmSize = 16 * 1024 * 1024
```
[the remark this answers is not in the document]
Comment (vuong177) [reply modules/light-clients/08-wasm/types/validation.go:3]: @misko how do you compile the grandpa contract? I've tried to compile the composable grandpa contract but the contract size still 1.6mb?

File: modules/light-clients/08-wasm/client/cli/query.go (hunk 3 above)
Comment (vuong177) [reply modules/light-clients/08-wasm/client/cli/query.go:68]: Added in newest commit. I forget to add this flag.

**Comment by vuong177** [comment]
@misko9 thanks for reviewing.

**Review by misko9 — APPROVED** [review]
LGTM!

**Pull request merged**